File: santander app/app/src/main/java/com/desafiosantander/DashBoard/DashBoardPresenter.java
package com.desafiosantander.DashBoard;

import android.support.annotation.NonNull;
import android.util.Log;

import java.lang.ref.WeakReference;
import java.util.ArrayList;
import java.util.Calendar;
import java.util.List;
import java.util.concurrent.TimeUnit;

interface DashBoardPresenterInput {
    public void presentDashBoardData(DashBoardResponse response);
    public void setStatmentList(List<Statement> statement);
}


public class DashBoardPresenter implements DashBoardPresenterInput {

    public static String TAG = DashBoardPresenter.class.getSimpleName();

    //weak var output: HomePresenterOutput!
    public WeakReference<DashBoardActivityInput> output;


    @Override
    public void presentDashBoardData(DashBoardResponse response) {
        // Log.e(TAG, "presentDashBoardData() called with: response = [" + response + "]");
        //Do your decoration or filtering here

    }

    @Override
    public void setStatmentList(List<Statement> statement) {
        DashBoardViewModel dashBoardViewModel = new DashBoardViewModel();
        dashBoardViewModel.statementList = statement;

        output.get().loadList(dashBoardViewModel);
    }


}

File: BankApp/app/src/main/java/com/accenture/bankapp/screens/dashboard/DashboardPresenter.kt
package com.accenture.bankapp.screens.dashboard

import timber.log.Timber
import java.lang.ref.WeakReference

interface DashboardPresenterInput {
    fun presentDashboardMetadata(dashboardResponse: DashboardResponse)
}

class DashboardPresenter : DashboardPresenterInput {
    var dashboardFragmentInput: WeakReference<DashboardFragmentInput>? = null

    override fun presentDashboardMetadata(dashboardResponse: DashboardResponse) {
        Timber.i("presentDashboardMetadata: Presenting Dashboard metadata")

        dashboardFragmentInput?.get()?.disableInfo()
        dashboardFragmentInput?.get()?.displayDashboardMetadata(DashboardViewModel(dashboardResponse.listStatements))
    }
}